playback/tape_cassettes/asynchronous/async_record_only_tape_cassette.py

```diff
@@ -61,6 +61,9 @@ def get_recording(self, recording_id):
     def iter_recording_ids(self, category, start_date=None, end_date=None):
         raise Exception("AsyncTapeCassette should only be used for recording, not playback")
 
+    def extract_recording_category(self, recording_id):
+        raise Exception("AsyncTapeCassette should only be used for recording, not playback")
```

Comment:
Exception is too broad - here and everywhere - it is bad practice

Reply:
The best fitting error I could find is TypeError, replaced with it.
From python doc:

exception TypeError
Raised when an operation or function is applied to an object of inappropriate type. The associated value is a string giving details about the type mismatch.

This exception may be raised by user code to indicate that an attempted operation on an object is not supported, and is not meant to be. If an object is meant to support a given operation but has not yet provided an implementation, NotImplementedError is the proper exception to raise.